test/pkg/integration/apis.go

```diff
@@ -520,22 +520,27 @@ func FindDBConfigFromPodEnv(componentName string, namespace string, client klien
 	}
 	pod := list.Items[0]
 
-	var password string
+	var password, host string
 	var port int32
-	var host string
 OuterLoop:
 	for _, c := range pod.Spec.Containers {
 		for _, v := range c.Env {
+			var findErr error
 			if v.Name == "DB_PASSWORD" {
-				password = v.Value
+				password, findErr = FindValueFromEnvVar(v, client, namespace)
 			} else if v.Name == "DB_PORT" {
-				pPort, err := strconv.ParseUint(v.Value, 10, 16)
+				var portStr string
+				portStr, findErr = FindValueFromEnvVar(v, client, namespace)
+				pPort, err := strconv.ParseUint(portStr, 10, 16)
```

Comment:
edge case: if `findErr != nil` it's likely that `password == ""` when will make `ParseUint` fail. In that case the "error parsing DB_PORT" error below would shadow `findErr`.
Possible solution: check for `findErr` immediately after making the call.

Reply:
Good eye, thank you! That seems like the best practice (immediately handle errors). Resolved in 475ef294fdb560756bd623f02c46df019dac9396.